Commands/Public/ping.js

```diff
@@ -0,0 +1,24 @@
+const { Stopwatch } = require("../../Modules/Utils/");
+
+module.exports = async ({ client, Constants: { Colors } }, documents, msg, commandData) => {
+	const timer = new Stopwatch();
+	await msg.send({
+		embed: {
+			color: Colors.INFO,
+			title: "Getting the ping 🏓",
+			description: "Please stand by...",
+		},
+	});
+	const sendPing = timer.duration;
+	timer.stop();
+	msg.send({
+		embed: {
+			color: Colors.LIGHT_GREEN,
+			title: "Pong! 🏓",
+			description: `Sending this message took ${Math.round(sendPing)}ms. The average hearbeat ping is ${Math.floor(client.ping)}ms`,
```

Comment:
But doesn't it literally only measure the time it takes to let DJS sent the message send request? bob isn't included, the timer only starts when the message is already received. Or does sending a message in DJS wait until it gets back its own message in an event?

Reply:
1. Bob will not send the message for sure.
2. Why? It's how long it takes for the message to send, assuming d.js isnt a potato and waits for events.
3. No it isn't really, assuming above. It's just a simple request.

Reply:
But where in that code is the user?!?!? lol.
It starts a timer, sends a message, stops the timer when it's sent. Done.

Reply:
I can just do it but I want to understand why lol.
I don't get what all that you just sent has to do with `new Stopwatch(); await msg.send(); timer.stop()`. Where in that part is the user typing `!ping`, that's not what the timer cares about, when that code is invoked the user has already sent the message, Discor dhas already gotten the message, it has already sent the gateway event which has already been parsed by DJS.

Reply:
No. User sends ping. User opens a socket and connects to the Discord API. Then the user sends a request. After that, the socket closes. Then Discord will parse the message and insert it to the database. Then it will broadcast the event to the open sockets. GAB's socket will catch the event. Then d.js will parse the JSON/ETF content and send the event. Then GAB does stuff with the message. After this, it will run the command. Again, it will create a socket and do the same steps as user did. Then it will edit the message, creating again a new socket, and again doing the same as user did.

Get your steps right Vlad.